### PR TITLE
Strategy switch bug

### PR DESCRIPTION
**Pull request by BioPhoton**

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -1776,10 +1776,10 @@
     yargs "^11.0.0"
     yargs-parser "10.0.0"
 
-"@rollup/plugin-commonjs@^14.0.0":
-  version "14.0.0"
-  resolved "https://registry.yarnpkg.com/@rollup/plugin-commonjs/-/plugin-commonjs-14.0.0.tgz#4285f9ec2db686a31129e5a2b415c94aa1f836f0"
-  integrity sha512-+PSmD9ePwTAeU106i9FRdc+Zb3XUWyW26mo5Atr2mk82hor8+nPwkztEjFo8/B1fJKfaQDg9aM2bzQkjhi7zOw==
+"@rollup/plugin-commonjs@^11.0.2":
```
Comment (BioPhoton) [inline yarn.lock:1779]: Please reset the yarn lock changes

File: libs/template/src/lib/core/render-aware/render-aware_creator.ts [hunk 2]
```diff
@@ -73,30 +77,30 @@ export function createRenderAware<U>(cfg: {
...
+          // Update completion, error and next
+          tap(cfg.updateObserver),
+          renderWithLatestStrategy(strategy$)
+        )
     ),
-    catchError(e => {
-      return EMPTY;
-    })
+    tap({
+      // TODO: doesnt work
```
Comment (kajetansw) [inline libs/template/src/lib/core/render-aware/render-aware_creator.ts:94]: Maybe this will help: https://github.com/BioPhoton/rx-angular/pull/199

Comment (BioPhoton) [reply libs/template/src/lib/core/render-aware/render-aware_creator.ts:94]: https://github.com/BioPhoton/rx-angular/pull/199#issuecomment-670229987

File: libs/template/src/lib/core/render-aware/render-aware_creator.ts [hunk 3]
```diff
@@ -108,9 +112,29 @@ export function createRenderAware<U>(cfg: {
     },
     activeStrategy$: strategy$,
     subscribe(): Subscription {
-      return new Subscription()
-        .add(strategy$.subscribe())
-        .add(renderingEffect$.subscribe());
-    }
+      return (
+        new Subscription()
+          //.add(strategy$.subscribe()) TODO: seems to work without, investigate
```
Comment (hoebbelsB) [inline libs/template/src/lib/core/render-aware/render-aware_creator.ts:117]: what about this one?

File: libs/template/src/lib/core/render-aware/render-aware_creator.ts [hunk 4]
```diff
@@ -108,9 +112,29 @@ export function createRenderAware<U>(cfg: {
...
+  strategyChanges$: Observable<RenderStrategy>
+): MonoTypeOperatorFunction<any> {
+  return (o$) => {
+    return o$.pipe(
+      withLatestFrom(strategyChanges$),
+      // hack to always use latest strategy on value change
+      switchMap(([renderValue, strategy]) =>
+        of(renderValue).pipe(
+          /*finalize(() => strategy.rxScheduleCD),
+             catchError(strategy.rxScheduleCD)*/
```
Comment (hoebbelsB) [inline libs/template/src/lib/core/render-aware/render-aware_creator.ts:134]:
```suggestion
```

**Pull request merged**